be/src/olap/rowset/beta_rowset.cpp

```diff
@@ -97,11 +101,13 @@ OLAPStatus BetaRowset::remove() {
 OLAPStatus BetaRowset::link_files_to(const std::string& dir, RowsetId new_rowset_id) {
     for (int i = 0; i < num_segments(); ++i) {
         std::string dst_link_path = segment_file_path(dir, new_rowset_id, i);
+        // TODO(lingbin): use Env API? or EnvUtil?
         if (FileUtils::check_exist(dst_link_path)) {
             LOG(WARNING) << "failed to create hard link, file already exist: " << dst_link_path;
             return OLAP_ERR_FILE_ALREADY_EXIST;
         }
         std::string src_file_path = segment_file_path(_rowset_path, rowset_id(), i);
+        // TODO(lingbin): how bos support link? use copy? or keep refcount to avoid being delete?
```

Comment:
What is bos? Can we change bos to external storage?

Reply:
Done.